[docs] Add a limitation for CDC in docs (#23586)
yugabyte/yugabyte-db: docs/content/preview/explore/change-data-capture/using-logical-replication/_index.md

- CDC is not supported on a target table for xCluster replication [11829](https://github.com/yugabyte/yugabyte-db/issues/11829).

> - Currently we don't support schema evolution for changes that require table rewrites (ex: ALTER TYPE). Additionally the operations DROP TABLE and TRUNCATE TABLE are not supported by CDC.

Reviewer: ```suggestion
- Currently, CDC doesn't support schema evolution for changes that require table rewrites (for example, ALTER TYPE), or DROP TABLE and TRUNCATE TABLE operations.
```
